generators/example/src/main/scala/ConfigMixins.scala

```diff
@@ -59,92 +63,58 @@ class WithUART extends Config((site, here, up) => {
     UARTParams(address = 0x54000000L, nTxEntries = 256, nRxEntries = 256))
 })
 
-// -----------------------------------------------
-// BOOM and/or Rocket Top Level System Parameter Mixins
-// -----------------------------------------------
 
-/**
- * Class to specify a "plain" top level BOOM and/or Rocket system
- */
-class WithTop extends Config((site, here, up) => {
-  case BuildTop => (clock: Clock, reset: Bool, p: Parameters) => {
-    Module(LazyModule(new Top()(p)).module)
-  }
+class WithNoGPIO extends Config((site, here, up) => {
+  case PeripheryGPIOKey => Seq()
 })
 
-/**
- * Class to specify a top level BOOM and/or Rocket system with DTM
- */
-class WithDTMTop extends Config((site, here, up) => {
-  case BuildTopWithDTM => (clock: Clock, reset: Bool, p: Parameters) => {
-    Module(LazyModule(new TopWithDTM()(p)).module)
+// DOC include start: tsi mixin
+class WithTSI extends Config((site, here, up) => {
+  case SerialKey => true
+  case BuildTop => (clock: Clock, reset: Bool, p: Parameters, success: Bool) => {
+    val top = up(BuildTop, site)(clock, reset, p, success)
+    success := top.connectSimSerial()
+    top
   }
 })
+// DOC include end: tsi mixin
 
-/**
- * Class to specify a top level BOOM and/or Rocket system with PWM
- */
-// DOC include start: WithPWMTop
-class WithPWMTop extends Config((site, here, up) => {
-  case BuildTop => (clock: Clock, reset: Bool, p: Parameters) =>
-    Module(LazyModule(new TopWithPWMTL()(p)).module)
+class WithDTM extends Config((site, here, up) => {
+  case BuildTop => (clock: Clock, reset: Bool, p: Parameters, success: Bool) => {
+    val top = up(BuildTop, site)(clock, reset, p, success)
+    top.reset := reset.asBool | top.debug.get.ndreset
```

Comment:
Is this circumventing reset synchronizers?

Reply:
Could you clarify? This was copied from the old DTMTop module, and I'm not familiar with some of its mechanics.

Reply:
Unless something's changed (which is possible), this is connecting the test-harness-level clock and reset to the DUT. That should be fine for simulation, but we might want to note that in a real chip you'd probably have an async top-level reset and a `ResetCatchAndSync` somewhere inside the chip. I was asking the question because I don't actually know if that `ResetCatchAndSync` is in the `RocketSubsystem` already or not (maybe @colinschmidt can chime in).

Reply:
It looks like Rocket Chip's TestHarness does 
```
  dut.reset := reset | dut.debug.map { debug => AsyncResetReg(debug.ndreset) }.getOrElse(false.B)
```

I guess we should be using the `AsyncResetReg` here?

Reply:
Yeah, we should fix that, but it wasn't the only problem I was referring to. The other thing was that you likely won't have a `clock` pin; you probably have a PLL, etc. that would generate a clock you need to synchronize `reset` to.

Reply:
So in Chipyard currently the hierarchy is `TestHarness -> Top`, while for real chips the hierarchy would likely look like `TestHarness -> ChipTop -> Top`. In BEAGLE the chip-specific clocking and reset needs are handled within `ChipTop`.

With the current BuildTop rules, it looks like the way to do this would be to call `p(BuildTop)` within `ChipTop`, creating something like:

```
class ChipTop(implicit val p: Parameters) extends RawModule {
  val sys_clk = Wire(Clock())
  val sys_rst = Wire(Bool())
  val sys_success = Wire(Bool())
  val sys = p(BuildTop)(sys_clock, sys_reset, p, sys_success)
  ...
}
```

The issue is currently we don't have this extra `ChipTop` layer within Chipyard. It seems like that would be a good thing to add in the future.

Reply:
I don't think rocket-chip currently does catch and sync for the top-level reset. They may be moving in that direction based on recent PRs, but its certainly not in our tagged version yet.
I like it would be cool to have something that is a "reasonable" `ChipTop` in chipyard so we can make the knowledge of how to do this "safely" more widely known.
I think thats outside of the scope of this PR, in that this doesn't change the current state of chipyard w.r.t. clock&reset. But we should file an issue to track a "realistic clock and reset mixin".

Reply:
I'll file an issue for `ChipTop`, while I'll leave it to someone more familiar with the clock&reset issues to write that.

https://github.com/ucb-bar/chipyard/issues/385